### PR TITLE
feat: Define the transition of windows and Prop for the views.

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
Make the standard configuration for navigations between windows.
Create a small example of how the navigation works.
Create a prop of when a user is logged in or not, different views can be shown.
Leave an example of this.
Create a public directory for images.

### CODE REVIEW

File: apps/frontend/ routes.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+const RutasNavegacion = {
+    HomePage: "/",
+    Routines: "/Routines",
+  };
+  
+  export default RutasNavegacion;
+
```
Comment (Miguelito2774) [inline apps/frontend/ routes.js]: add end-line please.

File: apps/frontend/ routes.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+const RutasNavegacion = {
```
Comment (Miguelito2774) [inline apps/frontend/ routes.js]: Rename this constant to English.

File: apps/frontend/app/AuthContextType.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [isLoggedIn, setIsLoggedIn] = useState(true);
+
+  const handleLogin = () => {
+    setIsLoggedIn(true);
+    document.cookie = "isLoggedIn=true; path=/";
+  };
+
```
Comment (Miguelito2774) [inline apps/frontend/app/AuthContextType.tsx:30]: Here should be a logout function that changes the state and deletes the cookie

File: apps/frontend/app/Header/NavBar.tsx [hunk 4]
```diff
@@ -2,12 +2,14 @@
 
 import { useState } from "react";
 import Image, { StaticImageData } from "next/image";
-import Icon from "./Icon"; 
 import SubMenu from "./SubMenu";
+import { ReactNode } from "react";
+import RutasNavegacion from "../../ routes";
```
Comment (Miguelito2774) [inline apps/frontend/app/Header/NavBar.tsx]: This import could be improved... make it cleaner.

File: apps/frontend/app/Header/NavBar.tsx [hunk 5]
```diff
@@ -32,30 +34,27 @@ const NavBar = ({
...
     setActiveIcon(id);
     console.log(`Icon clicked: ${id}`);
   };
 
-  const renderIcon = (src: StaticImageData, alt: string, onClick: () => void) => (
-    <Icon src={src} alt={alt} onClick={onClick} />
-  );
-
   return (
     <div className="bg-[#28292E]">
```
Comment (Miguelito2774) [inline apps/frontend/app/Header/NavBar.tsx]: 1. If you are using the div only to apply the background color, you can consider adding this color directly to the header, eliminating the div.
2.  Use semantic HTML.

File: apps/frontend/app/HomePage/ImageCard.tsx [hunk 6]
```diff
@@ -1,38 +1,45 @@
...
+  src: StaticImageData;
+  alt: string;
+  label: string;
+  onClickOption: string;
+}
+
+export default function ImageCard({ className, src, alt, label, onClickOption }: ImageCardProps) {
+  const router = useRouter(); 
 
   const handleImageClick = (text: string) => {
```
Comment (Miguelito2774) [inline apps/frontend/app/HomePage/ImageCard.tsx]: Do this component more flexible...
something like:

export default function ImageCard({ className, src, alt, label, onClickOption }: ImageCardProps) {
  const router = useRouter(); 

  const handleImageClick = () => {
    const route = RutasNavegacion[onClickOption]; 
    if (route) {
      router.push(route);
    } else {
      console.error("Route doesn't exist.");
    }
  };

**Review by Miguelito2774 — CHANGES_REQUESTED** [review]

File: apps/frontend/package.json [hunk 7]
```diff
@@ -12,7 +12,8 @@
     "next": "14.2.14",
     "next-pwa": "^5.6.0",
     "react": "^18",
-    "react-dom": "^18"
+    "react-dom": "^18",
```
Comment (Hairax) [inline apps/frontend/package.json]: What is the react-dom library being used for here?

File: apps/frontend/ routes.js (hunk 1 above)
Comment (GastonGutierrezC) [reply apps/frontend/ routes.js]: Done

File: apps/frontend/ routes.js (hunk 2 above)
Comment (GastonGutierrezC) [reply apps/frontend/ routes.js]: Done

File: apps/frontend/app/AuthContextType.tsx (hunk 3 above)
Comment (GastonGutierrezC) [reply apps/frontend/app/AuthContextType.tsx:30]: Done

File: apps/frontend/app/Header/NavBar.tsx (hunk 4 above)
Comment (GastonGutierrezC) [reply apps/frontend/app/Header/NavBar.tsx]: change the name of the route to make it clearer

File: apps/frontend/app/Header/NavBar.tsx (hunk 5 above)
Comment (GastonGutierrezC) [reply apps/frontend/app/Header/NavBar.tsx]: Done

File: apps/frontend/app/HomePage/ImageCard.tsx (hunk 6 above)
Comment (GastonGutierrezC) [reply apps/frontend/app/HomePage/ImageCard.tsx]: Done

File: apps/frontend/package.json (hunk 7 above)
Comment (GastonGutierrezC) [reply apps/frontend/package.json]: I am not using it at all, it was probably added by mistake.

**Review by Miguelito2774 — APPROVED** [review]

**Review by Hairax — APPROVED** [review]

**Pull request merged**